### ISSUE TITLE
Timeout values like `1s` result in number_format_exception

### ISSUE THREAD

**Issue by olivere** [elastic/elasticsearch#19075]
**Environment**

ES 5.0.0-alpha3
Linux/4.4.13-moby/amd64
Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_91/25.91-b14
running inside Docker.

**Problem**

I cannot issue "human" values for `timeout`any more, e.g. `timeout=1s`.

**Steps to reproduce**:

Notice the `"timeout":"1s"` here:

```
$ curl 'localhost:9200/_search' -d '{"query":{"match_all":{}},"timeout":"1s"}'
{"error":{"root_cause":[{"type":"number_format_exception","reason":"For input string: \"1s\""}],"type":"number_format_exception","reason":"For input string: \"1s\""},"status":400}
```

Notice the `"timeout":"1000"` here (same with `1000`, without the quotes btw):

```
$ curl 'localhost:9200/_search' -d '{"query":{"match_all":{}},"timeout":"1000"}'
{"took":1,"timed_out":false,"_shards":{"total":2,"successful":2,"failed":0},"hits":{"total":3,"max_score":1.0,"hits":[{"_index":"elastic-test","_type":"tweet","_id":"1","_score":1.0,"_timestamp":1466863101867,"_source":{"user":"olivere","message":"Welcome to Golang and Elasticsearch.","retweets":108,"created":"2012-12-12T17:38:34Z"}},{"_index":"elastic-test","_type":"tweet","_id":"2","_score":1.0,"_timestamp":1466863101879,"_source":{"user":"olivere","message":"Another unrelated topic.","retweets":0,"created":"2012-10-10T08:12:03Z"}},{"_index":"elastic-test","_type":"tweet","_id":"3","_score":1.0,"_timestamp":1466863101882,"_source":{"user":"sandrae","message":"Cycling is fun.","retweets":12,"created":"2011-11-11T10:58:12Z"}}]}}
```

Cannot find a deprecation notice. Is this expected behavior?

**Comment by dadoonet** [comment]
I don't think it's expected. I would actually expect the opposite: reject time values without unit.

**Comment by jasontedor** [comment]
> Cannot find a deprecation notice. Is this expected behavior?

This is a mistake, thanks for reporting. We require units on time values, so the behavior is completely the opposite of the documented behavior and the behavior in the 2.x series of Elasticsearch. I have opened #19077 to fix. Thanks for reporting! I have marked you as eligible for the [Pioneer Program](https://www.elastic.co/blog/elastic-pioneer-program).
